UKForeignOffice/lists
# Annual review scheduler: unpublish day actions

## Description

This feature adds support for the actions listed below for all lists where the `jsonData.currentAnnualReview` key is populated and the `jsonData.currentAnnualReview.keyDates.unpublish.UNPUBLISHED` key date is equal to or less than the current date:

- sends an email to all providers that have not confirmed or edited their provider record in response to the annual review email. This is detected by an Event record with the `EDIT` type greater than the annual review start date or an Event record  date with the `REMINDER` type greater than or equal to the `UNPUBLISHED` key date
- an Event record with type `REMINDER` is added for each list item where an email was sent 
- resets the annual review fields on all list items that have not confirmed or edited their provider record in response to the annual review email. Namely, `isAnnualReview` = false, `isPublished` = false, `status` = ANNUAL_REVIEW_OVERDUE
- an Event record is added for each list item with status `ANNUAL_REVIEW_OVERDUE`
- resets the annual review fields on the List record.  Namely, jsonData.currentAnnualReview is removed, `lastAnnualReviewStartDate` = nextAnnualReviewStartDate, `nextAnnualReviewStartDate` = nextAnnualReviewStartDate + 1 year 
- An Audit record is added with `eventName` = endAnnualReview to capture this event.

Note, a Prisma migration script has been added to insert a new ListItemEvent value `ANNUAL_REVIEW_OVERDUE`

## To test

- Set the annualReviewStartDate on the list to populate `List.jsonData.currentAnnualReview` 
- Run the scheduler batch process to populate `jsonData.currentAnnualReview.eligibleListItem`
- Using the /development page, set the annualReviewStartDate to the current date
- Run the scheduler worker process which will change the state on the list items (status = OUT_WITH_PROVIDER, isAnnualReview = true)
- Using the /development page, set the UNPUBLISHED key date to the current date or before
- Run the scheduler worker process which will perform the actions to unpublish the list and list items outlined above
- The banner on the list items page should be removed and any list items that were not responded to should be updated as UNPUBLISHED and REMOVED BY POST  

### Notes
The publishing status and activity status of the list items unpublished will appear as follows:
 
![image](https://user-images.githubusercontent.com/79845659/223401413-67b4fa98-532c-4aeb-a695-b07b54073dd1.png)

The last and next annual review start dates will be updated in the list page

![image](https://user-images.githubusercontent.com/79845659/223401970-1515e8f4-35ce-4617-b957-92e4fc4849bd.png)
